batchtable.py

```diff
@@ -39,12 +39,12 @@ def loadJSONBatch(self, data_in, object_wise = True):
 			n_objs = len(data_in)
 			# Find all the fields for all the objects
 			if type(data_in) is list:
-				data_in = {i: data_in[i] for i in xrange(len(data_in))}
-			for obj, objval in data_in.iteritems():
+				data_in = {i: data_in[i] for i in range(len(data_in))}
+			for obj, objval in iter(data_in.items()):
```

Comment:
Is there any particular reason to use `iter` here? I thought idiomatic Python 3 was just ot use `.items()` directly.

Reply:
That's what `2to3` does, but [PEP 469](https://www.python.org/dev/peps/pep-0469/) says `d.iteritems()` => `iter(d.items())`, so that's what I've been using.

Reply:
> Status: | Withdrawn

Reply:
```suggestion
			for obj, objval in data_in.items():
```

Reply:
Anyway, `data_in.items()` gives you a "view", which for the purposes of loops where you aren't updating the collection being traversed should behave like an iterator in terms of memory efficiency.

Reply:
Done in a2c2e00